### ISSUE TITLE
Log search function is very slow, sometimes causing system crash.

### ISSUE THREAD

**Issue by dangokuson** [sevdokimov/log-viewer#122]
Hi there! I am using `log viewer` to view log, `log viewer` will read all log files in `/default/logs` folder of the system. The system works stably except for the search function. The log file size can be up to 100MB, I'm not sure if this is the main cause of the error. However, when I use the log search function, the system may crash, sometimes. 

Can someone tell me what is the problem?

**Comment by sevdokimov** [comment]
100MB is not too big a file, it should work fast.  Do you have `java.lang.OutOfMemoryError`?

**Comment by dangokuson** [comment]
@sevdokimov No, I don't. But, there are a lot of file logs under `/default/logs` folder. It is around 700 files, and a sub-folder, that sub-folder also contains 700 log files. However, those files were compressed files. The crash application error occur only when I use the search log function.

Are there any suggestions?

**Comment by sevdokimov** [comment]
Do you open files one by one? or all files at the same time in one view?

**Comment by dangokuson** [comment]
@sevdokimov I did open several files at the same time, but the maximum is 10 files. I used the below function for search:

![image](https://user-images.githubusercontent.com/86276600/163165488-25c74f66-4c5b-43ee-9e2d-2dccc7bb12b4.png)

**Comment by sevdokimov** [comment]
Are there any errors in the browser console or log-viewer output?

**Comment by dangokuson** [comment]
Nope, there is no error and the application has been frozen for a few minutes before crash

**Comment by sevdokimov** [comment]
What exactly crashes, a page in Brauser or the java application (backend)?
